packages/dds/merge-tree/docs/Obliterate.md

```diff
@@ -0,0 +1,639 @@
+
+## Eventual Consistency Strategy
+
+This section is focused on how one could implement the "obliterate" semantics inside merge tree in an eventually consistent fashion.
+This will constitute the bulk of the complexity of the feature.
+Since obliterate is generally a "different kind of remove," there may be a nice abstraction to introduce at the code level to generalize
+removal information. However, in favor of introducing niceties later this design document will assume fields are inlined and focus on
+the strategy for ensuring eventual consistency. If such an abstraction is introduced, ideally it would enable better "pay-to-play" of
+common code paths based on merge-tree feature usage.
+As an example, `BaseSegment.split()` needs to copy segment properties to the split segment.
+So new properties added to segment will unnecessarily copy undefined values.
+
+There are a few aspects of merge tree's bookkeeping and general feature set that require consideration when designing new op semantics:
+
+- Any changes to direct fields of tree nodes themselves (either new data or changes to bookkeeping of existing data)
+- How the feature interacts with an increasing collab window and zamboni
+- Impact on the partial lengths scheme
+- Bookkeeping and handling of overlapping removals (note some may be obliterates and some may not be)
+- Reconnection
+- Snapshotting impact
+
+We'll first present an overview of a potential scheme for implementing the obliterate op, then comment on these aspects.
+
+### High-level bookkeeping changes
+
+Segments will be augmented with `movedSeq` and `localMovedSeq` fields which generally align with the semantics of `seq, localSeq, removedSeq,` and `localRemovedSeq`.
+When segments are moved and not just obliterated, they will also contain a reference to the destination segment.
+This may look as follows:
+
+```typescript
+/**
+ * Tracks information about when and where this segment was moved to.
+ * @example - Suppose a merge tree had 3 TextSegments "X", "A", and "B", and
+ * received the operation `move({ start: 0, end: 1 }, { dest: 3 }, { seq: 30 })` (moving the "X"
+ * after the "A" and the "B").
+ * After processing this operation, it would have the segments `[<moved "X" tombstone>, "A", "B", "X"]`.
+ * The moved "X" tombstone segment would have the following IMoveInfo: `{ movedSeq: 30, moveDst: <reference to living "X" segment>}`
+ */
+export interface IMoveInfo {
+    /**
+     * Local seq at which this segment was moved if the move is yet-to-be acked. Only set on the tombstone "source" segment of the move.
+     */
+    localMovedSeq?: number;
+    /**
+     * Seq at which this segment was moved. Only set on the tombstone "source" segment of the move.
+     */
+    movedSeq: number;
+    /**
+     * A reference to the inserted destination segment corresponding to this segment's move.
+     * If undefined, the move was an obliterate.
+     */
+    moveDst?: ReferencePosition;
```

Comment:
this might be a good place to use a tracking group, as it supports one to many, specifically if the dest splits we be able to keep track of all the destination segments